packages/runtime/datastore/src/channelContext.ts

```diff
@@ -69,12 +69,14 @@ export function createServiceEndpoints(
     };
 }
 
-export function summarizeChannel(
+// TODO: this needs to be split up into two calls so callers do all the captures first before summarizeState
+export async function summarizeChannel(
     channel: IChannel,
     fullTree: boolean = false,
     trackState: boolean = false,
-): ISummaryTreeWithStats {
-    const summarizeResult = channel.summarize(fullTree, trackState);
+): Promise<ISummaryTreeWithStats> {
+    const state = channel.captureSummaryState(fullTree);
+    const summarizeResult = await channel.summarizeState(state);
```

Comment:
I'm a bit at a loss here.
You have added IChannel.captureRevision().summarize() flow.
Why is there some other captureSummaryState() flow? I do not see IChannel having captureSummaryState()
Is this simply non-updated code from prior revisions?

Reply:
Yes, non-updated. The last couple commits were just to get feedback on the change to SharedObject before changing it everywhere.